.github/workflows/release.yml

```diff
@@ -0,0 +1,267 @@
+          **/target/failsafe-reports/*
+
+    # Record the release-version in git and add the git tag for the release.
+    - name: Record ${{ github.event.inputs.releaseVersion }} release in git
+      env:
+        RELEASE_VERSION: ${{ github.event.inputs.releaseVersion }}
+        GIT_TAG: nessie-${{ github.event.inputs.releaseVersion }}
+      run: |
+        git commit -a -m "[release] release nessie-${RELEASE_VERSION}"
+        git tag ${GIT_TAG}
+
+    # Update versions to next development iteration and add a new git commit
+    - name: Set next development version version ${{ github.event.inputs.nextVersion }}
+      env:
+        NEXT_VERSION: ${{ github.event.inputs.nextVersion }}
+      run: |
+        ./mvnw versions:set -DgenerateBackupPoms=false -DnewVersion=${NEXT_VERSION}-SNAPSHOT
+        (cd python ; bumpversion --no-commit --new-version ${NEXT_VERSION} minor)
+
+        git commit -a -m "[release] next development iteration ${NEXT_VERSION}-SNAPSHOT"
+
+    # Push the 2 git commits and git tag. If this one fails, some other commit was pushed to the
+    # 'main' branch and break the linear history for the Nessie git repo.
+    - name: Push tag + branch
+      env:
+        GIT_TAG: nessie-${{ github.event.inputs.releaseVersion }}
+      run: |
+        UPSTREAM="$(git remote)"
+
+        git push ${UPSTREAM}
+        git push ${UPSTREAM} :refs/tags/${GIT_TAG}
+        git push ${UPSTREAM} ${GIT_TAG}
+
+    # This last step finally deploys the artifacts. All build and test steps were already ran
+    # in previous steps
+    # TODO close + deploy the OSSRH staging repo, maybe switch from maven-deploy-plugin to nexus-staging-maven-plugin to allow that?
+    - name: Publish artifacts for release ${{ github.event.inputs.releaseVersion }}
+      env:
+        RELEASE_VERSION: ${{ github.event.inputs.releaseVersion }}
+        GIT_TAG: nessie-${{ github.event.inputs.releaseVersion }}
+        MAVEN_USERNAME: ${{ secrets.OSSRH_ACCESS_ID }}
+        MAVEN_OSSRH_TOKEN: ${{ secrets.OSSRH_TOKEN }}
+        MAVEN_GPG_PASSPHRASE: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
+        SPARK_LOCAL_IP: localhost
+      run: |
+        # Reset to workspace to release tag (the next version has been configured)
+        git reset --hard ${GIT_TAG}
+
+        mvn -B deploy --file pom.xml -Pnative -Prelease -DdeployAtEnd=true -DskipTests
+
+        (cd tools/apprunner-gradle-plugin/ ; ./gradlew publishPlugins)
+
+        (cd python ; python -m twine upload dist/*)
+
+        echo '${{ secrets.DOCKER_TOKEN }}' | docker login -u '${{ secrets.DOCKER_USERNAME }}' --password-stdin
+        docker push projectnessie/nessie:${RELEASE_VERSION} && docker push projectnessie/nessie:latest
+
+        git changelog -t ${RELEASE_VERSION} -n -x /dev/null | egrep -v '^  [*] (Bump |\[release\] ).*$' > /tmp/release-notes.txt
+        gh release create ${GIT_TAG} --notes-file /tmp/release-notes.txt --title "Nessie version ${RELEASE_VERSION}"
```

Comment:
I think we would either need to do this by hand or add an extra template. eg we want to push a bit more than just the changelog in the release notes. we also push highlights and download locations for the artifacts etc. 

So its templating a non-small markdown doc